### ISSUE TITLE
slideshow request

### ISSUE THREAD

**Issue by 1941Jan** [darktable-org/darktable#12929]
I would like to be able to show the captions in the slideshow. Now I can use a program that is only available in windows and it has some problems I can't resolve. 
I will be very happy if that is possible. 

Jan Minekus
algemeen@minekus.org

  **Is your feature request related to a problem? Please describe.**

**Describe the solution you'd like**

**Alternatives**

**Additional context**

**Comment by pehar1** [comment]
I also would like to have this functionality. It should be configurable which information is displayed and at which location this should happen.

@1941Jan , did you try _XnView MP_. It is available for Linux, Windows, MacOS and you can create highly configurable slideshows.

**Comment by dterrahe** [comment]
This sounds like applying an additional style, like in export, while doing a slideshow. In this case containing a watermark.

It would help to make the subject of the issue more descriptive of the request.

**Comment by pehar1** [comment]
Yes, I thought of something like that. But of course I do not know the intention of @1941Jan who opened this feature request. @1941Jan , can you please specify more precisely what **you** have in mind and adjust the subject (title)?

**Comment by TurboGit** [comment]
I'm not sure about the watermark which will be displayed over the image. We may want to have some data on a screen corner?

**Comment by dterrahe** [comment]
I was just thinking that would allow full SVG functionality, which would be useful for more professional presentations (but likely there are other, better tools for that). I'm not sure it is currently possibly to extend images ("reverse crop") so that a full landscape area covering the whole screen can be used for the watermark even in case of portrait images.

**Comment by TurboGit** [comment]
> For example in a collection of art photo's it would be nice to see the artist and may be the title.

And where would you like to see those information? On a corner? Over the image? Other idea...

**Comment by dterrahe** [comment]
> it is currently possibly to extend images ("reverse crop")

Obviously I was thinking of the _framing_ module. Select aspect 16:9 and the image would cover the whole screen. Then you can place an (svg) watermark wherever you want; on the image or just outside it. Add both framing and watermark in a style and apply it automatically when doing a slideshow. (there actually seems to be a bug in _framing_ where if you set border size to 0, the aspect ratio is not enforced)

#10025 may also be somewhat relevant here.

**Comment by 1941Jan** [comment]
The program should be doing just what I want, Thank you!
I installed it and following the help instruction I can create a slideshow.
However I have not been able to add a caption.
I have to look for more information to discover why.

Jan Minekus

**Comment by SilvioGrosso** [comment]
Hello, 
+1. This would be really useful in my view.

Nothing fancy or overly complicated to implement.
For my workflow, just a simple caption, at the botton of the current picture (in the slideshow), which allows to *automatically* display the name of the picture, would be enough.

**Comment by pehar1** [comment]
> For my workflow, just a simple caption, at the botton of the current picture (in the slideshow), which allows to automatically display the name of the picture, would be enough.

My intention is a bit different. I would like to show the _description_ of the image at the bottom. This description is sometimes very short (one or two words), sometimes it is a longer text with several sentences. So some kind of centered (block) formatting would be required.

Another form of implementation I can imagine, without using a style with a svg, is a configurable "overlay" similar to the one in the lighttable view.

- Put this "overlay" in a freely movable window
- Make the size of this window adjustable
- Make the font size adjustable
- ...

and we have a very flexible extension of the slideshow.

**Comment by dterrahe** [comment]
> we have a very flexible extension of the slideshow.

Point is, it will not be as flexible as watermark, which allows the "complete" freedom of an svg template using tags and other metadata, and it would still be a lot of work to implement. Or if it _is_ as flexible, it would basically be a _duplication_ of watermark.

> My intention is a bit different

That's the point; everybody will want something slightly different. And they usually want _what is currently possible_ + _this small easy additional thing_. 

We know this _now_. implementing something limited just means it will need to be done _twice_ to get it right. And we'll probably end up with a confusing set of choices.

**Comment by pehar1** [comment]
> they usually want what is currently possible + this small easy additional thing.

I have been programming myself for several decades, first Pascal, then Modula2, now Java. I am fully aware that seemingly small change requests can mean huge programming work. Unfortunately, my programming skills in C are zero. Otherwise I would most likely have been actively involved in the programming of darktable myself. But C is an insurmountable hurdle for me at the moment...

> it will not be as flexible as watermark

I can also imagine the watermark concept very well. My idea also first went in this direction. Unfortunately, my knowledge of the source code is far from sufficient to judge which type of implementation might be the "best" or the "easiest". Please understand my suggestions as a constructive attempt of a user with half-knowledge to contribute to the project, nothing more.

**Comment by TurboGit** [comment]
@pehar1: You are a lucky guy! You've been using some sane languages all derived from Modula. I've also used Modula2 & Pascal then a lot of Ada.

**Comment by 1941Jan** [comment]
I think it would be best at the bottom of the photo, overlapping the 
bottom part.
Even better if it does not overlap part of the photo.

Jan Minekus

**Comment by dterrahe** [comment]
Here we go again. It is not in the best interest of the project as a whole for limited solutions to be implemented by people with little knowledge of the code and the potential to leverage components to achieve more powerful, consistent, functionality. Such code still needs to be reviewed and maintained by the core developers, so it is likely to cost them _more_ time in the end and it might only help the original contributor and no-one else with slightly different requirements. Or feature creep and restructurings might eventually bring it to the same point after much more time spent.

Please try to think of this feature request in more generic terms and edit the title to reflect those so this is more likely to be picked up by an interested dev. Of which there might be more now that slideshow got such an overhaul.

I still believe, but this is without looking at the code as well, that following the same approach as already exists in _export_, allowing an additional style to be specified that will be applied during slideshow, will give the best bang for the buck. This way, using the framing and watermark modules, which have received more testing than stand-alone functionality would, you could add flexible texts with backgrounds and even logos. Additional sample svgs could be added that work especially well with slideshows and provide examples that users could customise. Possibly a sample style could be provided as well.

**Comment by parafin** [comment]
Having such additional style applied will make any existing (e.g. pre-generated from console) cache invalid. So IMHO a bad idea.

**Comment by dterrahe** [comment]
> So IMHO a bad idea.

I had not thought of that. Does the pipeline not use cached partial results and then apply (changed) later modules on top of them?

I guess we'll end up with a limited solution that's not useful to many users then.

**Comment by parafin** [comment]
> I had not thought of that. Does the pipeline not use cached partial results and then apply (changed) later modules on top of them?

Not sure, especially since slideshow has been reworked. I was thinking of thumbnail cache. Again, watermark is not exactly the last step in the pipeline.

**Comment by dterrahe** [comment]
> watermark is not exactly the last step in the pipeline.

Isn't it? It is the top most module in my list; thought that meant last?

**Comment by parafin** [comment]
I think there is hidden gamma on top

**Comment by jenshannoschwalm** [comment]
> Does the pipeline not use cached partial results and then apply (changed) later modules on top of them?

That's the pixelpipe cache ...

**Comment by dterrahe** [comment]
I thought the rebuilt slideshow was using a pixelpipe. I'm afraid I don't understand much of this stuff as I've never really looked into it.

**Comment by jenshannoschwalm** [comment]
Yes the new code works in a pixelpipe. But there is no caching involved as I understand the code.

**Comment by parafin** [comment]
If there’s no caching at all and it’s fast enough, it meand my objection is nullified.

**Comment by TurboGit** [comment]
But again as we can't display a watermark outside of the image boundary I'm not sure that's the way to go.

**Comment by dterrahe** [comment]
If you switch on the framing module with 16:9 aspect and landscape, it will expand the image to fill the screen. Then you can put a watermark wherever you want. Putting those two together in a style is not complicated. And you still have the flexibility to want the text (or whatever watermark) only _inside_ the image (adjusted for portrait/landscape) by _not_ using framing. Building all those options separately (and everything else that watermark + svg offers) would in my eyes just be a massive duplication.

Now it might make sense to support "thumbnail overlays" _also_ in slideshow. For those who just want a really simple way to see filename, but without a lot of configuration options.
